### PR TITLE
Project: Memory optimizations

### PR DESCRIPTION
**Pull request by REASY**
# Optimal way to write string to the buffer:
## Before:
image

## After:
image

---
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/607)

### CODE REVIEW

**Comment by REASY** [comment]
# [DeadHeadingStats.java](src/main/java/beam/analysis/plots/DeadHeadingStats.java) can be improved by passing attributes with event:
## Before
image
image

## After:
image

File: src/main/java/beam/utils/BeamCalcLinkStats.java [hunk 1]
```diff
@@ -160,35 +160,52 @@ public void writeFile(final String filename) {
                         Link link = this.network.getLinks().get(linkId);
 
                         out.write(linkId.toString());
-                        out.write("," + link.getFromNode().getId().toString());
-                        out.write("," + link.getToNode().getId().toString());
+                        // out.write("," + link.getFromNode().getId().toString());
```
Comment (JustinPihony) [inline src/main/java/beam/utils/BeamCalcLinkStats.java]: Nit: I'd say remove all the comments

File: src/main/java/beam/utils/BeamCalcLinkStats.java [hunk 2]
```diff
@@ -160,35 +160,52 @@ public void writeFile(final String filename) {
...
+                        writeCommaAndStr(out, link.getFromNode().getId().toString());
```
Comment (JustinPihony) [inline src/main/java/beam/utils/BeamCalcLinkStats.java]: What are your thoughts on removing the `out` via an implicit? Maybe hiding too much and would make it harder to debug for others?

Comment (REASY) [reply src/main/java/beam/utils/BeamCalcLinkStats.java]: @JustinPihony , as far as I know there is no `implicits` in Java. Or you want to extract this to Scala method with `implicit` parameter?

**Pull request merged**